Mathlib/CategoryTheory/Localization/DerivabilityStructure/OfFunctorialResolutions.lean

```diff
@@ -0,0 +1,131 @@
+have functorial right resolutions, then this is a morphism of localizers
+in the other direction. -/
+@[simps]
+def localizerMorphismInv [W₂.HasTwoOutOfThreeProperty] :
+    LocalizerMorphism W₂ W₁ where
+  functor := ρ
+  map := by
+    rw [hW₁]
+    intro X Y f hf
+    have := i.naturality f
+    dsimp at this
+    simp only [MorphismProperty.inverseImage_iff]
+    rw [← W₂.precomp_iff _ _ (hi X), ← this]
+    exact W₂.comp_mem _ _ hf (hi Y)
+
+variable [Φ.functor.Full] [Φ.functor.Faithful]
+
+variable (i) in
+/-- If `Φ : LocalizerMorphism W₁ W₂` corresponds to a class `W₁` that is
+induced by `W₂` via the fully faithful functor `Φ.functor` and that we
+have functorial right resolutions given by a functor `ρ : C₂ ⥤ C₁`, then
+this is the natural transformation `𝟭 C₁ ⟶ Φ.functor ⋙ ρ` induced
+by `i : 𝟭 C₂ ⟶ ρ ⋙ Φ.functor`. -/
+noncomputable def ι : 𝟭 C₁ ⟶ Φ.functor ⋙ ρ :=
+  ((whiskeringRight C₁ C₁ C₂).obj Φ.functor).preimage (whiskerLeft Φ.functor i)
+
+@[simp]
+lemma Φ_functor_map_ι_app (X₁ : C₁) :
+    Φ.functor.map ((ι i).app X₁) = i.app (Φ.functor.obj X₁) :=
+  NatTrans.congr_app (((whiskeringRight C₁ C₁ C₂).obj Φ.functor).map_preimage
+    (X := 𝟭 C₁) (Y := Φ.functor ⋙ ρ) (whiskerLeft Φ.functor i)) X₁
+
+include hW₁ hi in
+lemma W₁_ι_app (X₁ : C₁) : W₁ ((ι i).app X₁) := by
+  simpa only [hW₁, MorphismProperty.inverseImage_iff, Φ_functor_map_ι_app]
+    using hi (Φ.functor.obj X₁)
+
+end functorialRightResolutions
+
+variable [Φ.functor.Full] [Φ.functor.Faithful] [W₂.HasTwoOutOfThreeProperty]
+
+open functorialRightResolutions
+include hi hW₁
+
+lemma isLocalizedEquivalence_of_functorial_right_resolutions :
+    Φ.IsLocalizedEquivalence :=
+  Φ.isLocalizedEquivalence_of_unit_of_unit (localizerMorphismInv hi hW₁) (ι i) i
+    (W₁_ι_app hi hW₁) hi
+
+variable [W₂.IsMultiplicative]
+
+lemma isConnected_rightResolution_of_functorial_resolutions (X₂ : C₂) :
+    letI : W₁.IsMultiplicative := by rw [hW₁]; infer_instance
+    IsConnected (Φ.RightResolution X₂) := by
+  have : W₁.IsMultiplicative := by rw [hW₁]; infer_instance
+  have : Nonempty (Φ.RightResolution X₂) := ⟨{ hw := hi X₂, .. }⟩
+  have : IsPreconnected (Φ.RightResolution X₂) :=
+    zigzag_isPreconnected (fun R₀ R₄ ↦ by
+      let R₁ : Φ.RightResolution X₂ := { hw := W₂.comp_mem _ _ R₀.hw (hi _), .. }
+      let R₂ : Φ.RightResolution X₂ := { hw := hi X₂, .. }
+      let R₃ : Φ.RightResolution X₂ := { hw := W₂.comp_mem _ _ R₄.hw (hi _), .. }
+      let f₀ : R₀ ⟶ R₁ := { hf := W₁_ι_app hi hW₁ R₀.X₁, .. }
+      let f₁ : R₂ ⟶ R₁ :=
+        { f := ρ.map R₀.w
+          comm := (i.naturality R₀.w).symm
+          hf := (localizerMorphismInv hi hW₁).map _ R₀.hw }
+      let f₂ : R₂ ⟶ R₃ :=
+        { f := ρ.map R₄.w
+          comm := (i.naturality R₄.w).symm
+          hf := (localizerMorphismInv hi hW₁).map _ R₄.hw }
+      let f₃ : R₄ ⟶ R₃ := { hf := W₁_ι_app hi hW₁ R₄.X₁, .. }
+      exact (Zigzag.of_hom f₀).trans ((Zigzag.of_inv f₁).trans
```

Comment:
FYI, in #25405, I propose to register a `Trans` instance on Zigzag, so that this can become a `calc` proof if you want.

Reply:
The calc block was just a suggestion as I was adding the instance. If you feel like it was more readable before, it’s your call.